sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/EvaluationOutcome.java

```diff
@@ -24,6 +22,7 @@
 
     private final Type type;
     private final String source;
+    private final MesosResource mesosResource;
```

Comment:
Happy to in another PR.  This is adding a single field.  I like the two `pass` and the one `fail` method.  Reads better in the code I think. I get that the `null` arguments in some places isn't great.  What would a builder alternative look like though?

```
return new Evaluation.Builder(this, true, "Reason of thing because %s, %s", reasonA, reasonB)
    .setMesosResource(mesosResource)
    .build();
```
vs
```
return pass(this,  null, "Reason of thing because %s, %s", reasonA, reasonB);
```

Here.  I'm adopting the rule, which we should follow everywhere that a builder shouldn't be constructable unless an immediate `build()` call generates a valid object.  We violate this rule all over the place, but I'd like to stop.